Exercise 9 doesn't compile

Exercise 9 doesn't compile with the given solution, not sure if intended as the book doesn't mention this.

```bash
error[E0600]: cannot apply unary operator `-` to type `u32`
  --> exercises/09_ambiguity_and_ordering/main.rs:50:36
   |
50 |         NumberType::NegativeNumber(-$negative)
   |                                    ^^^^^^^^^^ cannot apply unary operator `-`
...
70 |     get_number_type!(-5).show();
   |     -------------------- in this macro invocation
   |
   = note: unsigned values cannot be negated
   = note: this error originates in the macro `get_number_type` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0600`.
error: could not compile `macrokata` due to previous error
```

--- a/exercises/09_ambiguity_and_ordering/main.rs
+++ b/exercises/09_ambiguity_and_ordering/main.rs
@@ -8,7 +8,7 @@ enum NumberType {
     /// The user wrote a literal, positive number.
     PositiveNumber(u32),
     /// The user wrote a literal, negative number.
-    NegativeNumber(u32),
+    NegativeNumber(i32),
     /// We can't tell if the user's code is positive or negative because it's a block.
     UnknownBecauseBlock(i32),
     /// We can't tell if the user's code is positive or negative because it's an expression.

